src/Microsoft.ComponentDetection.Detectors/maven/MvnPomCliComponentDetector.cs

```diff
@@ -0,0 +1,112 @@
+namespace Microsoft.ComponentDetection.Detectors.Maven;
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Threading.Tasks;
+using System.Xml;
+using Microsoft.ComponentDetection.Contracts;
+using Microsoft.ComponentDetection.Contracts.BcdeModels;
+using Microsoft.ComponentDetection.Contracts.Internal;
+using Microsoft.ComponentDetection.Contracts.TypedComponent;
+using Microsoft.Extensions.Logging;
+
+public class MvnPomCliComponentDetector : FileComponentDetector
+{
+    public MvnPomCliComponentDetector(
+     IComponentStreamEnumerableFactory componentStreamEnumerableFactory,
+     IObservableDirectoryWalkerFactory walkerFactory,
+     ILogger<MvnCliComponentDetector> logger)
+    {
+        this.ComponentStreamEnumerableFactory = componentStreamEnumerableFactory;
+        this.Scanner = walkerFactory;
+        this.Logger = logger;
+    }
+
+    public override string Id => "MvnPomCli";
+
+    public override IList<string> SearchPatterns => new List<string>() { "*.pom" };
+
+    public override IEnumerable<string> Categories => new[] { Enum.GetName(typeof(DetectorClass), DetectorClass.Maven) };
+
+    public override IEnumerable<ComponentType> SupportedComponentTypes => new[] { ComponentType.Maven };
+
+    public override int Version => 2;
+
+    protected override async Task OnFileFoundAsync(ProcessRequest processRequest, IDictionary<string, string> detectorArgs)
+    {
+        await this.ProcessFileAsync(processRequest);
+    }
+
+    private async Task ProcessFileAsync(ProcessRequest processRequest)
+    {
+        var singleFileComponentRecorder = processRequest.SingleFileComponentRecorder;
+        var stream = processRequest.ComponentStream;
+
+        try
+        {
+            byte[] pomBytes = null;
+
+            if ("*.pom".Equals(stream.Pattern, StringComparison.OrdinalIgnoreCase))
```

Comment:
This check is not necessary. Only files that match the `SearchPatterns` will call `ProcessFileAsync`